app/ts/schema.ts

```diff
@@ -0,0 +1,41 @@
+import * as funtypes from 'funtypes'
+import { ParsedValueConfig } from 'funtypes/lib/types/ParsedValue'
+
+export const BigIntHexSchema = funtypes.String.withParser({
+	parse(value) {
+		if (!/^0x([a-fA-F0-9]{1,64})$/.test(value)) return { success: false, message: `${value} is not a hex string encoded number.` } satisfies funtypes.Failure
+		return { success: true, value: BigInt(value) } satisfies funtypes.Success<bigint>
+	},
+	serialize(value) {
+		if (typeof value !== 'bigint') return { success: false, message: `${typeof value} is not a bigint.` } satisfies funtypes.Failure
+		return { success: true, value: `0x${value.toString(16)}` } satisfies funtypes.Success<string>
+	},
+	name: 'BigInt',
+})
+
+export const TokenSchema = funtypes.Object({
+	chainId: BigIntHexSchema,
+	name: funtypes.String,
+	address: funtypes.String,
+	symbol: funtypes.String,
+	decimals: BigIntHexSchema,
+})
+
+export const TokenListSchema = funtypes.Array(TokenSchema)
+
+export const ManagedTokensCacheParserConfig: ParsedValueConfig<funtypes.String, funtypes.Static<typeof TokenListSchema>> = {
+	parse(value) {
+		const jsonParsed = JSON.parse(value)
+		return TokenListSchema.safeParse(jsonParsed)
+	},
+	serialize(value) {
+		const serializedValue = TokenListSchema.safeSerialize(value)
+		if (!serializedValue.success) return { success: false, message: serializedValue.message }
+		const jsonString = JSON.stringify(serializedValue.value)
+		return { success: true, value: jsonString }
+	},
+}
+
+export const ManagedTokensCacheSchema = funtypes.String.withParser(ManagedTokensCacheParserConfig)
+
+export type ManagedTokensCache = funtypes.Static<typeof ManagedTokensCacheSchema>
```

Comment:
```suggestion

```
Shouldn't need these, TokenListSchema.serialize() should work fine as is

Reply:
this is actually needed to stringify the data before sending to localStorage unlike StorageArea which accepts objects

Reply:
```ts
let x: TokenListSchema
const stringifiedContent = x.safeSerialize() // The value should be string